### DISCUSSION TITLE
Unable to push to github using https. error: RPC failed; curl 56 Send failure

### DISCUSSION THREAD

**Discussion by extorn** [community/community#22836] [New to GitHub]
<p>I’ve been trying all morning to push my local repo to github to no avail.<br>
I’ve tried pushing in android studio, using git command line (in cmd and gitbash), and now using git desktop.<br>
I’ve tried increasing the http.postbuffer size as suggested in similar reports and I’ve checked the git.config file to confirm it is set  (“postBuffer = 524288000”). This doesn’t fix it.</p>
<p>I’m trying to push using https because the ssh method seems to fail authentication every time. I’ve checked my ssh-agent is running and contains the private key, but this doesn’t seem to have made any difference. I’ve also checked that the ssh public key is added to my github profile under ssh keys.</p>
<p>The error I get is consistent and as you can see this is for a small repository:</p>
<p>Enumerating objects: 21675, done.<br>
Counting objects: 100% (21675/21675), done.<br>
Delta compression using up to 16 threads<br>
Compressing objects: 100% (5543/5543), done.<br>
Writing objects: 100% (21675/21675), 42.65 MiB | 189.05 MiB/s, done.<br>
Total 21675 (delta 15235), reused 21096 (delta 14925), pack-reused 0<br>
error: RPC failed; curl 56 Send failure: Connection was aborted<br>
fatal: the remote end hung up unexpectedly<br>
fatal: the remote end hung up unexpectedly<br>
Everything up-to-date</p>

**Comment by extorn** [comment, marked as answer]
<p>For anyone else experiencing this:<br>
Solution:<br>
My issue is that my service provider is “virgin media”. They have something called “websafe”. I enable this to protect my child. The way this seems to work is by mitm. I presume this because to fix the error I have done the following:</p>
<ol>
<li>Open my virgin media account in a browser</li>
<li>navigate to the websafe settings</li>
<li>navigated to the “websites section”</li>
<li>added “<a href="http://github.com" rel="noopener nofollow ugc">github.com</a>” as an “Always allowed” website.</li>
<li>Attempted connection using ssh to github (from powershell)</li>
<li>Received a “Are you sure you want to continue connecting (yes/no/[fingerprint])? yes” message.</li>
<li>Said yes - the repo is now uploading absolutely fine using ssh (which has never worked before).<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2e92485b874dcddb863a3345dea970ea690ab1c1.jpeg" data-download-href="/uploads/short-url/6DZn6UtLsQUmQMGzvjq8mDfuorT.jpeg?dl=1" title="screenshot_virgin" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/28184810/181073326-f6a0828e-23be-496f-81f1-2833f18c5ac4.jpeg" alt="screenshot_virgin" data-base62-sha1="6DZn6UtLsQUmQMGzvjq8mDfuorT" width="690" height="319" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e92485b874dcddb863a3345dea970ea690ab1c1_2_690x319.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e92485b874dcddb863a3345dea970ea690ab1c1_2_1035x478.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e92485b874dcddb863a3345dea970ea690ab1c1_2_1380x638.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e92485b874dcddb863a3345dea970ea690ab1c1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">screenshot_virgin</span><span class="informations">2975×1378 268 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ol>

**Comment by dbass-clark** [comment]
<p>I am trying to push a Unity3D project up to Github repo and I am getting a similar error. Also using Unity3D. Tried command line and the GUI for Github. Also tried adding LFS data (50 GB) and tried creating several different repositories. Seems to be an issue with large file sizes? The file I am trying to push is about 3 GB. Here is what I get:</p>
<p>Enumerating objects: 8805, done.<br>
Counting objects: 100% (8805/8805), done.<br>
Delta compression using up to 16 threads<br>
Compressing objects: 100% (6970/6970), done.<br>
error: RPC failed; curl 55 Send failure: Connection was aborted<br>
fatal: the remote end hung up unexpectedly<br>
Writing objects: 100% (8804/8804), 2.99 GiB | 6.18 MiB/s, done.<br>
Total 8804 (delta 4902), reused 3380 (delta 1833), pack-reused 0<br>
fatal: the remote end hung up unexpectedly<br>
Everything up-to-date</p>

**Comment by extorn** [comment]
<p>If you read the github faq, it says that it only supports git “push” operations of sizes up to 150mb (or something like this, I’m recalling from when I was investigating my issue).<br>
They apparently have some different method for dealing with large files. You’re actually pushing lots of smaller files totaling 3GB from what I can see above. Others have suggested breaking the push into sections by pushing just some commits then a next batch of commits. (presumably in historical order). I’d try that perhaps.<br>
My issue was different, I couldn’t even push a single push containing multiple commits with a total size of 40MB.</p>

**Comment by airtower-luna** [comment]
<p>The push and file size limits are described here: <a href="https://docs.github.com/en/github/managing-large-files/conditions-for-large-files">Conditions for large files</a></p>
<p>At the moment it says maximum 50 MB per file, and 100 MB in a single push. If you have larger files and want to switch to LFS, note that you’ll have to <a href="https://docs.github.com/en/github/managing-large-files/removing-files-from-a-repositorys-history">remove them from history too</a> to make the push to Github work.</p>

**Comment by gits-gits** [comment]
It’s not working for me at all please help to resolve this issue

**Reply by gits-gits** [reply]
I am migrating code from perforce and file size is 63gb and my push always fails don’t know how to fix it , please suggest something . I tried buffer size change and lfs still no luck

**Reply by airtower-luna** [reply]
That is an order of magnitude above the recommended *maximum repository size*, total, and two orders of magnitude above the allowed size of a single push: https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github

I've had the misfortune of dealing with Perforce for work a few years back, so I remember it encourages putting everything into one "depot". If you have different projects in your Perforce, it'd make to split it into one repository per project. For large individual files you might want to consider LFS (also described in the documentation linked above).
